torch/_inductor/lowering.py

```diff
@@ -377,12 +377,23 @@ def inner(*inputs: List[TensorBox], alpha=None):
                 inputs[-1] = mul(inputs[-1], alpha)
         else:
             assert alpha is None
+
+        # Get the first TensorBox/ExpandView input as the reference tensor for
+        # sizes, data types, and device.
```

Comment:
`promote_constants` handles this for `ir.Constant` by wrapping it in an `ExpandView`:
https://github.com/pytorch/pytorch/blob/51c2e22e94caf56ac89967093b274ac81c7ca9e2/torch/_inductor/lowering.py#L351-L353

I think we could just do the same for `IndexingConstant` below:
https://github.com/pytorch/pytorch/blob/51c2e22e94caf56ac89967093b274ac81c7ca9e2/torch/_inductor/lowering.py#L356